src/components/layout/SignMessageLayout.tsx

```diff
@@ -19,39 +20,43 @@ export const SignMessageLayout = ({
   message = undefined,
   buttonText = 'Sign message',
 }: Props): JSX.Element => {
-  const { address, isConnecting } = useAccount()
+  const { address, isConnected } = useAccount()
 
   return (
-    <div className="w-full">
-      <div className="flex h-screen w-full flex-col">
-        <div className="w-full p-5 text-2xl font-bold">
-          <FontAwesomeIcon icon={faPrayingHands} size="1x" className="m-2" />
-        </div>
-        <div className="md:bg-warm-gray-50 m-auto space-y-8 rounded-none border-none bg-none p-8 py-8 shadow-none dark:bg-slate-900 md:rounded-lg md:border md:border-solid md:p-4 md:shadow-sm">
-          {children}
-
-          {isConnecting && !address && !message ? (
-            <LoaderSpinner />
-          ) : (
-            <ConnectButton
-              accountStatus="full"
-              showBalance={false}
-              chainStatus={'none'}
-            />
-          )}
+    <Web3Provider>
+      <div className="w-full">
+        <div className="flex h-screen w-full flex-col">
+          <div className="w-full p-5 text-2xl font-bold">
+            <FontAwesomeIcon icon={faPrayingHands} size="1x" className="m-2" />
+          </div>
+          <div className="md:bg-warm-gray-50 m-auto space-y-8 rounded-none border-none bg-none p-8 py-8 shadow-none dark:bg-slate-900 md:rounded-lg md:border md:border-solid md:p-4 md:shadow-sm">
+            {children}
 
-          {address && message && (
-            <div className="flex w-full items-center justify-center">
-              <SignMessageButton
-                text={buttonText}
-                message={message}
-                onSignSuccess={onSignSuccess}
-                onSignError={(): void => void toast.error('Linking bot denied')}
+            {isConnected && !address && !message ? (
```

Comment:
You should either check for `isConnected` or `!address`, you don't need to check both
```suggestion
            {isConnected && !message ? (
```